tools/mpremote/tests/test_filesystem.sh

```diff
@@ -170,3 +170,20 @@ EOF
 $MPREMOTE resume cp -r "${TMP}/package" :
 $MPREMOTE resume ls : :package :package/subpackage
 $MPREMOTE resume exec "import package; package.x(); package.y()"
+
+# Test rm -r functionality
+echo -----
+$MPREMOTE resume mkdir :testdir
+$MPREMOTE resume cp -r "${TMP}/package" :testdir/package
+$MPREMOTE resume ls :testdir
+$MPREMOTE resume rm -r :testdir/package
+$MPREMOTE resume ls :testdir
+
+# Test rm -r on non-existent path
+echo -----
+$MPREMOTE resume rm -r :nonexistent || echo "expect error"
+
+# Test rm -r on read-only filesystem
+echo -----
+$MPREMOTE resume exec "import os; os.mount(os.VfsFat(RAMBlockDev(512, 50)), '/readonly', readonly=True)"
```

Comment:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 19] ENODEV
```

The RAMBlockDev object requires an mkfs call on it before it can successfully be mounted.

There's been some movement in the vfs API recently; make sure you are running the full test harness against a firmware version you built from your exact commit, not just some other release firmware version.